packages/features/ee/common/server/LicenseKeyService.ts

```diff
@@ -94,4 +99,16 @@ class LicenseKeyService {
   }
 }
 
+export class NoopLicenseKeyService implements ILicenseKeyService {
+  async incrementUsage(_usageEvent?: UsageEvent): Promise<any> {
+    // No operation
+    return Promise.resolve();
+  }
+
+  async checkLicense(): Promise<boolean> {
```

Comment:
NOOP is only Instantiated when there is no license key - we can auto return false here.